pyo3-benches/benches/bench_frompyobject.rs

```diff
@@ -23,20 +23,22 @@ fn enum_from_pyobject(b: &mut Bencher<'_>) {
 
 fn list_via_downcast(b: &mut Bencher<'_>) {
     Python::with_gil(|py| {
-        let any: &PyAny = PyList::empty(py).into();
+        let any: &Bound<'_, PyAny> = &PyList::empty_bound(py);
 
         b.iter(|| {
-            let _list: &PyList = black_box(any).downcast().unwrap();
+            let list = black_box(any).downcast::<PyList>().unwrap();
+            black_box(list)
         });
     })
 }
 
 fn list_via_extract(b: &mut Bencher<'_>) {
     Python::with_gil(|py| {
-        let any: &PyAny = PyList::empty(py).into();
+        let any: &Bound<'_, PyAny> = &PyList::empty_bound(py);
 
         b.iter(|| {
-            let _list: &PyList = black_box(any).extract().unwrap();
+            let list: Bound<'_, PyList> = black_box(any).extract().unwrap();
+            black_box(list)
         });
     })
```

Comment:
I'm hoping that these two benchmarks (which are often volatile and a bit noisy) are more stable after this change, we'll have to see...

Reply:
Returning the list is certainly worthwhile but I don't think the black box around the return value is necessary IIRC.

Reply:
You're correct, the return value from the closure is already black-boxed inside `criterion`.

My goal here was to ensure that the `.extract().unwrap()` isn't optimised out. I'll give this another pass to tidy up.